core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -714,6 +715,16 @@ package object config {
       .booleanConf
       .createWithDefault(true)
 
+  private[spark] val SHUFFLE_SERVICE_DB_BACKEND =
+    ConfigBuilder(Constants.SHUFFLE_SERVICE_DB_BACKEND)
+      .doc("Specifies a disk-based store used in shuffle service local db. " +
+        "Only LEVELDB is supported now.")
+      .version("3.4.0")
+      .stringConf
+      .transform(_.toUpperCase(Locale.ROOT))
+      .checkValues(DBBackend.values.map(_.toString).toSet)
+      .createWithDefault(DBBackend.LEVELDB.name)
```

Comment:
do we want this to be truly pluggable where user could just provide their own class name and we just use?   If all the API's are public seems like that is the only difference.  Other than perhaps documentation.   I'm ok either way I guess just don't want to have to add another config or change this api if we want to go that way.

Reply:
My idea is to provide only one of two ways to use it, and I don't want the user to specify the implementation class

Reply:
The purpose of doing this is to use RocksDB more smoothly(like SHS using RocksDB),  I have never thought of making it completely pluggable.